server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -567,10 +571,10 @@ private static Map<String, DiscoveryNodeRole> rolesToMap(final Stream<DiscoveryN
     private static Map<String, DiscoveryNodeRole> roleMap = rolesToMap(DiscoveryNodeRole.BUILT_IN_ROLES.stream());
 
     public static DiscoveryNodeRole getRoleFromRoleName(final String roleName) {
-        if (roleMap.containsKey(roleName) == false) {
-            throw new IllegalArgumentException("unknown role [" + roleName + "]");
+        if (roleMap.containsKey(roleName)) {
```

Comment:
While experimenting, run into interesting issue: the role name are case-sensitive, for example:

```
node.roles: [ master, Master]
```

In general, it was fine, OpenSearch would refuse to start because `Master` was not recognized as a valid role. You may already see where I am going: with the dynamic roles, such set of roles becomes totally valid and confusing:

```
curl http://localhost:9200/_cat/nodes
127.0.0.1 14 84 7 2.24 2.35 3.06 m Master,master * aredko-ThinkPad-P15-Gen-1
```

I think we should be a bit more strict now with the way roles are treated and checks should be case-insensitive: in the example above `master` and `Master` should be reduced to same role `master`. @ylwu-amzn wdyt?

Reply:
@ylwu-amzn can I have you look over this?

Reply:
The whole case-sensitive thing can be dealt with as a separate issue IMO.